pkg/types/openstack/validation/machinepool.go

```diff
@@ -20,5 +23,26 @@ func ValidateMachinePool(p *openstack.MachinePool, fldPath *field.Path) field.Er
 		}
 	}
 
+	if len(p.AdditionalNetworkIDs) > 0 {
+		for i, uuid := range p.AdditionalNetworkIDs {
+			if !isValidUUID(uuid) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalNetworkIDs"), p.AdditionalNetworkIDs, fmt.Sprintf("Invalid Network ID #%d", i)))
+			}
+		}
+	}
+
+	if len(p.AdditionalSecurityGroupIDs) > 0 {
+		for i, uuid := range p.AdditionalSecurityGroupIDs {
+			if !isValidUUID(uuid) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalSecurityGroupIDs"), p.AdditionalSecurityGroupIDs, fmt.Sprintf("Invalid Security Group ID #%d", i)))
```

Comment:
`fldPath.Child("additionalSecurityGroupIDs").Index(i)`

Reply:
`p.AdditionalSecurityGroupIDs` -> `uuid` in the `field.Invalid`

Reply:
`Invalid Security Group ID %#d` -> `a valid UUID V4 must be provided`

Reply:
all this same for the `AdditionalNetworkIDs`

Reply:
also this could be easily DRyed up
```go
func ValidateMachinePool(p *openstack.MachinePool, fldPath *field.Path) field.ErrorList {
..
..
    allErrs = append(allErrs, validateUUIDV4s(p.AdditionalNetworkIDs, fldPath.Child("additionalNetworkIDs")...)
    allErrs = append(allErrs, validateUUIDV4s(p.AdditionalSecurityGroupIDs, fldPath.Child("additionalSecurityGroupIDs")...)

}
func validateUUIDV4s(input []string, fldPath field.Path) field.ErrorList {
    allErrs := field.ErrorList{}
        for idx, uuid := range input {
            if !validUUID(uuid) {
                allErrors = append(allErrors, field.Invalid(fldPath.Index(idx), uuid, "valid UUID v4 must be specified"))
            }
      }
}
```

Reply:
Done